`master` broken with camlp5 7.06

#### Version: `master`
#### Description of the problem
With `camlp5` 7.06 and OCaml `4.07.0+rc2` we have:

```
File "plugins/ltac/g_obligations.ml", line 65, characters 2-18:
Error: This function has type
         'weak2 Pcoq.Gram.entry -> 'weak2 Extend.extend_statement -> unit
       It is applied to too many arguments; maybe you forgot a `;'.
Makefile.build:703: recipe for target 'plugins/ltac/g_obligations.cmo' failed
make[1]: *** [plugins/ltac/g_obligations.cmo] Error 2
make[1]: *** Waiting for unfinished jobs....
File "plugins/ltac/g_ltac.ml4", line 83, characters 8-11049:
Error: This function has type
         'weak1 Pcoq.Gram.entry -> 'weak1 Extend.extend_statement -> unit
       It is applied to too many arguments; maybe you forgot a `;'.
Makefile.build:703: recipe for target 'plugins/ltac/g_ltac.cmo' failed
make[1]: *** [plugins/ltac/g_ltac.cmo] Error 2
File "plugins/ltac/g_rewrite.ml", line 814, characters 2-18:
Error: This function has type
         'weak2 Pcoq.Gram.entry -> 'weak2 Extend.extend_statement -> unit
       It is applied to too many arguments; maybe you forgot a `;'.
Makefile.build:703: recipe for target 'plugins/ltac/g_rewrite.cmo' failed
make[1]: *** [plugins/ltac/g_rewrite.cmo] Error 2
make[1]: Leaving directory '/home/egallego/tmp/coq-master'
Makefile:163: recipe for target 'submake' failed
make: *** [submake] Error 2
```
This doesn't happen in `v8.8`.

Same here (noticed because this breaks Iris CI), with OCaml 4.05.0.

```
OCAMLC    plugins/ltac/g_obligations.ml
File "plugins/ltac/g_obligations.ml4", line 60, characters 8-373:
Error: This function has type
         'a Pcoq.Gram.entry -> 'a Extend.extend_statement -> unit
       It is applied to too many arguments; maybe you forgot a `;'.
Makefile.build:703: recipe for target 'plugins/ltac/g_obligations.cmo' failed
make[1]: *** [plugins/ltac/g_obligations.cmo] Error 2

OCAMLC    plugins/ltac/g_rewrite.ml
File "plugins/ltac/g_rewrite.ml4", line 206, characters 8-86:
Error: This function has type
         'a Pcoq.Gram.entry -> 'a Extend.extend_statement -> unit
       It is applied to too many arguments; maybe you forgot a `;'.
Makefile.build:703: recipe for target 'plugins/ltac/g_rewrite.cmo' failed
make[1]: *** [plugins/ltac/g_rewrite.cmo] Error 2

File "plugins/ltac/g_ltac.ml4", line 83, characters 8-11049:
Error: This function has type
         'a Pcoq.Gram.entry -> 'a Extend.extend_statement -> unit
       It is applied to too many arguments; maybe you forgot a `;'.
Makefile.build:703: recipe for target 'plugins/ltac/g_ltac.cmo' failed
make[1]: *** [plugins/ltac/g_ltac.cmo] Error 2
```

--- a/grammar/coqpp_main.ml
+++ b/grammar/coqpp_main.ml
@@ -250,7 +250,7 @@ let print_rule fmt r =
 let print_entry fmt gram e =
   let print_position_opt fmt pos = print_opt fmt print_position pos in
   let print_rules fmt rules = print_list fmt print_rule rules in
-  fprintf fmt "let () =@ @[%s.safe_extend@ %s@ @[(%a, %a)@]@]@ in@ "
+  fprintf fmt "let () =@ @[%s.gram_extend@ %s@ @[(%a, %a)@]@]@ in@ "
     gram e.gentry_name print_position_opt e.gentry_pos print_rules e.gentry_rules
 
 let print_ast fmt ext =

--- a/parsing/pcoq.ml
+++ b/parsing/pcoq.ml
@@ -336,7 +336,7 @@ module Gram =
       I'm not entirely sure it makes sense, but at least it would be more correct.
           *)
       G.delete_rule e pil
-    let safe_extend e ext = grammar_extend e None ext
+    let gram_extend e ext = grammar_extend e None ext
   end
 
 (** Remove extensions

--- a/parsing/pcoq.mli
+++ b/parsing/pcoq.mli
@@ -80,7 +80,7 @@ module type S =
   (* Get comment parsing information from the Lexer *)
   val comment_state : coq_parsable -> ((int * int) * string) list
 
-  val safe_extend : 'a entry -> 'a Extend.extend_statement -> unit
+  val gram_extend : 'a entry -> 'a Extend.extend_statement -> unit
 
   (* Apparently not used *)
   val srules' : production_rule list -> symbol

